packages/frontend/app/lib/api/tenant.server.ts

```diff
@@ -0,0 +1,163 @@
+import { gql } from '@apollo/client'
+import type {
+  CreateTenantInput,
+  CreateTenantMutation,
+  CreateTenantMutationVariables,
+  QueryTenantsArgs,
+  ListTenantsQuery,
+  ListTenantsQueryVariables
+} from '~/generated/graphql'
+import { getApolloClient } from '../apollo.server'
+
+export const listTenants = async (request: Request, args: QueryTenantsArgs) => {
+  const apolloClient = await getApolloClient(request)
+  const response = await apolloClient.query<
+    ListTenantsQuery,
+    ListTenantsQueryVariables
+  >({
+    query: gql`
+      query ListTenantsQuery(
+        $after: String
+        $before: String
+        $first: Int
+        $last: Int
+      ) {
+        tenants(after: $after, before: $before, first: $first, last: $last) {
+          edges {
+            node {
+              id
+              email
+              apiSecret
+              idpConsentUrl
+              idpSecret
+              publicName
+              createdAt
+              deletedAt
+              isOperator
+            }
+          }
+          pageInfo {
+            startCursor
+            endCursor
+            hasNextPage
+            hasPreviousPage
+          }
+        }
+      }
+    `,
+    variables: args
+  })
+  return response.data.tenants
+}
+
+export const createTenant = async (
+  request: Request,
+  args: CreateTenantInput
+) => {
+  const apolloClient = await getApolloClient(request)
+  const response = await apolloClient.mutate<
+    CreateTenantMutation,
+    CreateTenantMutationVariables
+  >({
+    mutation: gql`
+      mutation CreateTenantMutation($input: CreateTenantInput!) {
+        createTenant(input: $input) {
+          tenant {
+            id
+            publicName
+            email
+            apiSecret
+            idpConsentUrl
+            idpSecret
+          }
+        }
+      }
+    `,
+    variables: {
+      input: args
+    }
+  })
+
+  return response.data?.createTenant
+}
+
+export const updateTenant = async (
+  request: Request,
+  args: UpdateTenantInput
```

Comment:
looks like this and many other types are showing ts errors... look like they just need to be imported? run `pnpm --filter frontend typecheck` or see the frontend ci job failing.

Reply:
@BlairCurrey sorry about that, I have not pushed some of my files (including the logging). 
Sorted.